shared/test/TestTileSource.cpp

```diff
@@ -0,0 +1,95 @@
+#include "CoordinateSystemFactory.h"
+#include "DefaultSystemToRenderConverter.h"
+#include "SymbolAnimationCoordinatorMap.h"
+#include "Tiled2dMapSource.h"
+#include "VectorTileGeometryHandler.h"
+#include "WebMercatorTiled2dMapLayerConfig.h"
+#include "helper/GeoJsonGenerator.h"
+
+#include "helper/TestScheduler.h"
+#include <algorithm>
+#include <atomic>
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_get_random_seed.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <thread>
+#include <vector>
+
+struct TestLoaderResult final {
+    std::optional<std::string> etag = std::nullopt;
+    LoaderStatus status = LoaderStatus::OK;
+    std::optional<std::string> errorCode = std::nullopt;
+};
+
+class TestTiled2dMapVectorSource : public Tiled2dMapSource<int, std::shared_ptr<TestLoaderResult>, int> {
+  public:
+    TestTiled2dMapVectorSource(const std::shared_ptr<TestScheduler> scheduler)
+        : Tiled2dMapSource(MapConfig(CoordinateSystemFactory::getEpsg3857System()),
+                           std::make_shared<WebMercatorTiled2dMapLayerConfig>(
+                               "mock", "{z}/{x}/{y}", Tiled2dMapZoomInfo(1.0, 0, 0, false, true, false, true), 0, 20),
+                           CoordinateConversionHelperInterface::independentInstance(), scheduler, 62, 1, "layer") {};
+
+    std::unordered_set<Tiled2dMapTileInfo> getCurrentTiles() {
+        std::unordered_set<Tiled2dMapTileInfo> tiles;
+        for (const auto &tile : currentTiles) {
+            tiles.insert(tile.first);
+        }
+        return tiles;
+    }
+
+    void notifyTilesUpdates() override {}
+
+  protected:
+    void cancelLoad(Tiled2dMapTileInfo tile, size_t loaderIndex) override {}
+
+    ::djinni::Future<std::shared_ptr<TestLoaderResult>> loadDataAsync(Tiled2dMapTileInfo tile, size_t loaderIndex) override {
+        auto promise = std::make_shared<::djinni::Promise<std::shared_ptr<TestLoaderResult>>>();
```

Comment:
AFAIU, this does not have to be make_shared-ed, just `::djinni::Promise<std::shared_ptr<TestLoaderResult>> promise;` suffices.